activerecord/test/support/method_stub.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module MethodStub
+  private
+
+    def with_stubbed_method(object, method_name, return_value = nil)
```

Comment:
I'm not sure this would be accurate, what about when we pass `SomeObject.singleton_class` as the object argument?

Reply:
Currently, I have not the right answer. We might discuss it later on new short Pull Request
as I mentioned:
> Would be great to see simple PR with https://github.com/rails/rails/pull/33162/files#diff-81e98371a331e686a926985a7b9e9d88 activerecord/test/support/method_stub.rb file. It might help concentrate more people on it and get more suggestions with possible improvements to it.